etc/workflows/studio-upload.xml

```diff
@@ -142,7 +142,7 @@
         exception-handler-workflow="partial-error"
         description="Tagging source material for archival">
       <configurations>
-        <configuration key="source-flavors">dublincore/*,security/*,*/source</configuration>
+        <configuration key="source-flavors">dublincore/*,security/*,smil/*,*/source</configuration>
```

Comment:
I'm not sure I follow. If I put the `editor` operation in `partial-transcode-studio-tracks`, that happens before `prepare-av`, but it operates on the media generated by `prepare-av`. I mean, maybe it doesn't have to, I'm not sure, but that's the way it is in the standard workflow, where I stole this from. :sweat_smile: 

And what's the reasoning behind archiving the trimmed version without the cutting information as opposed to all of the source material, like the standard workflow does? :thinking:

Reply:
The idea behind the `partial-transcode-studio-tracks` is not to loose the source material at any state of the workflow but archive the transcoded version of the tracks without waste space with original video files. 

1. mark source material for archive (but do not take a snapshot)
1. transcode all tracks
1. switch the flavors and archive tags
1. take snapshot and go on with regular processing

If any operation from 1. or 2. fail, the error handler workflow will archive the original source files and you will be save. 
Ideally you want to replace the transcode operation (2.) with the editor. If it does not work (for reasons), put the editor operation in between 2. and 3.

As other workflows expects the `*/prepared` version, this must be generated and archived. Otherwise the following workflows will fail. In most cases all tracks does have an audio stream and the prepared version is same as source. AssetManager keep track of it and make a hardlink on filesystem.

Reply:
Okay, I think I understand now. However, I'm not sure I agree with your premise, that we actually want to archive the trimmed version, and just discard the original videos.

That's not what we do for other ways of getting data into Opencast, either. When a CA uploads a video and someone cuts it, we then also don't just throw away the uncut version.

On the other hand when Studio finally learns to cut the video itself client side, we would also not have the uncut version in Opencast, right? Or would we still upload the uncut version plus a SMIL file and only use the client side trimming for the download version? :thinking: 

Can we get a third (or even more for all I care :wink:) opinion on this?

Reply:
The initial trimming is initiated by the user, that creates the video file with studio and decide to trim it. It has nothing to do with the editor in the admin UI. We should respect it.

Reply:
Right now, I would slightly tend to keep the original file as well (i.e. uncut), as this is unlikely to be significantly larger than the cut one (Studio users will very likely only remove tens of seconds at most). However, as mentioned, as soon as Studio actually does the cutting itself, Opencast does not have the uncut version anyway. And I would certainly oppose deliberately uploading the uncut version from Studio. 

Soooooo... "I don't care" is what I want to say :D I think it doesn't make a large difference and the situation will change in some time anyway.

Reply:
Personally I'm very much in the camp where the complete set of original files is kept, and trimmed copies are treated as any other intermediate file.  We want a single source of truth, and discarding the original after trimming it seems wrong to me.

That being said, those saying that the user trimmed the video prior to ingesting it, and that we should respect that do have a point.  What if the user is trimming off something that's not allowed to be kept (say, privacy reasons, or licensing).  Do our adopters run a risk by potentially having that footage in their archive, whether they know they have it or not?  If the trim in Studio was truly done client side (this would be a nice feature), then we wouldn't get the data at all - and we would be ok with that.  So why do we feel like we get to keep it when the presentr thinks that we're not even getting it in the first place.  Normally I would say that the presenter is out of luck, but *especially* in light of the current COVID-19 situation we have to assume that these recordings may be being done in suboptimal places.  Kids could be run through the frame, someone could spill their coffee, all kinds of things which would not otherwise occur in a classroom or office setting.

As much as I prefer in all other cases to keep the original source media and create an intermediate trimmed copy, I think _in this case_ we need to respect user choice.  Perhaps we should document that this is not what is done in other cases in the studio docs, so that institutions can make the choice to keep everything if they really feel a need for it.  It will also (hopefully) help us in the future when someone eventually digs this up and asks why we did it differently :D